### DISCUSSION TITLE
Modern.js v2 发布：支持 Rspack 构建

### DISCUSSION THREAD

**Discussion by chenjiahan** [web-infra-dev/modern.js#3183] [Announcements]
大家好，很高兴地向大家宣布，Modern.js v2 版本已经正式发布了！

**Modern.js 是字节跳动 Web Infra 团队开源的一套 Web 工程体系**。在开源以来的一年多时间里，Modern.js 保持稳定的迭代节奏，数十位贡献者参与了开发，累计提交 2000+ 个 Pull Request，并支持了 Rspack 构建、嵌套路由、流式渲染等新特性。

在这篇文章里，我们会和大家一起聊一聊 Modern.js 在过去一年多时间里的变化。

点击查看正文：[《Modern.js v2 发布：支持 Rspack 构建》](https://modernjs.dev/blog/updates/v2-release-note.html)

**Comment by zmzlois** [comment]
Hey nice work y'all! 

I turned on `ssr: true` in `modern.config.ts` and it didn't seem like it supports `experimental-rspack` at the moment?

**Comment by chenjiahan** [comment]
Yes, currently `ssr: true` is partially supported. 

If you don't need to use nested routes, you can enable SSR + Rspack. But when you use SSR + Rspack + nested routes, there are still some problems. 

For example, the `loadable-components-webpack-plugin` is required for SSR, but it does not work in Rspack yet. We opened an issue [#2704](https://github.com/web-infra-dev/rspack/issues/2704) for Rspack yesterday and it has been resolved.

Perhaps @GiveMe-A-Name can provide more context on this.
